src/core/packages/chrome/browser-internal/src/ui/project/sidenav_v2/navigation/to_navigation_items.tsx

```diff
@@ -429,30 +430,80 @@ const isRecentlyAccessedDefinition = (
 const filterEmpty = <T,>(arr: Array<T | null | undefined>): T[] =>
   arr.filter((item) => item !== null && item !== undefined) as T[];
 
-function warnAboutDuplicateIcons(logoItem: SideNavLogo, primaryItems: MenuItem[]) {
-  // Collect all items with icons
-  const itemsWithIcons = [logoItem, ...primaryItems]
-    .filter((item) => item.iconType && item.iconType !== FALLBACK_ICON)
-    .map((item) => ({ id: item.id, icon: String(item.iconType) }));
-
-  // Group items by icon and warn about duplicates
-  const iconGroups = new Map<string, string[]>();
-  itemsWithIcons.forEach(({ id, icon }) => {
-    const items = iconGroups.get(icon) || [];
-    iconGroups.set(icon, [...items, id]);
+/**
+ * Generic function to detect and warn about duplicate values in navigation items.
+ * @param values - Array of values to check for duplicates
+ * @param formatWarning - Function to format the warning message for duplicates
+ */
+function warnAboutDuplicates(
+  values: string[],
+  formatWarning: (value: string, count: number) => string
+) {
+  const valueGroups = new Map<string, number>();
+  values.forEach((value) => {
+    valueGroups.set(value, (valueGroups.get(value) || 0) + 1);
   });
 
-  iconGroups.forEach((itemIds, iconType) => {
-    if (itemIds.length > 1) {
-      warnOnce(
-        `Icon "${iconType}" is used by multiple navigation items: ${itemIds.join(
-          ', '
-        )}. Consider using unique icons for better UX.`
-      );
+  valueGroups.forEach((count, value) => {
+    if (count > 1) {
+      warnOnce(formatWarning(value, count));
     }
   });
 }
 
+function warnAboutDuplicateIcons(logoItem: SideNavLogo, primaryItems: MenuItem[]) {
+  if (SKIP_WARNINGS) return;
+  // Collect all items with icons (only logo + primary items, excluding fallback)
+  const icons = [logoItem, ...primaryItems]
+    .filter(
+      (item) =>
+        item.iconType && item.iconType !== FALLBACK_ICON && typeof item.iconType === 'string'
+    )
+    .map((item) => String(item.iconType));
+
+  warnAboutDuplicates(
+    icons,
+    (icon) =>
+      `Icon "${icon}" is used by multiple navigation items. Consider using unique icons for better UX.`
+  );
+}
+
+function warnAboutDuplicateIds(
+  logoItem: SideNavLogo,
+  primaryItems: MenuItem[],
+  footerItems: MenuItem[]
+) {
+  if (SKIP_WARNINGS) return;
+  // Collect all IDs from all items, including secondary menu items
+  let allIds: string[] = [logoItem.id];
+
+  // Helper to extract IDs from menu items including their secondary sections
+  const collectIds = (items: MenuItem[]) => {
+    items.forEach((item) => {
+      allIds.push(item.id);
+      if (item.sections) {
+        item.sections.forEach((section) => {
+          allIds.push(section.id);
+          section.items.forEach((secondaryItem) => {
+            allIds.push(secondaryItem.id);
+          });
+        });
+      }
+    });
+  };
```

Comment:
Not super important, just polish - could probably be moved outside of the `warnAboutDuplicateIds` body into a utility function that returns `allIds`